action.php

```diff
@@ -281,6 +281,10 @@ protected function sortRules($rules)
                 if ($partA === '*') return 1;
                 if ($partB === '*') return -1;
 
+                // regex goes after what matches it
+                if ($this->containsRegex($partA) && preg_match("/$partA/", $partB)) return 1;
+                if ($this->containsRegex($partB) && preg_match("/$partB/", $partA)) return -1;
```

Comment:
This looks weird to me. you're applying the regex in one path segment to the path segment it is compared to? I'm not sure I understand what this is trying to do. Can you explain?

Reply:
Let's say we have two rules (this exact case is taken from the included unit test):

```
reg:12345:(sub-\d{3}):*
reg:12345:sub-678:bus
```

Here we have a regex pattern and an exact string that would match the regex. When sorting the rules according to significance, I want to make sure that the exact match comes before the regex in the rule stack, as it is more specific.

Or can you think of a more robust test case in `\helper_plugin_aclplusregex_test::providerSortRules` that would help me refine the code?